internal/artifacts/fetch.go

```diff
@@ -113,12 +114,18 @@ func (m *Manager) fetchImageByDigest(digestRef name.Digest, architecture Arch, i
 	// verify the image signature, we only accept properly signed images
 	logger.Debug("verifying image signature")
 
-	_, bundleVerified, err := cosign.VerifyImageSignatures(ctx, digestRef, &m.options.ImageVerifyOptions)
-	if err != nil {
+	var verifyFailedCount int = 0
```

Comment:
this might better be extracted to a separate sub-function which verifies, and verification should go until the first successful verification returning the result

Reply:
Done